Packs/BitcoinAbuse/Integrations/BitcoinAbuse/BitcoinAbuse.py

```diff
@@ -0,0 +1,269 @@
+        api_key: str
+
+    Returns:
+        report_address_params: _ReportAddressParams if validation succeeded
+        raises DemistoException if validation failed
+
+    """
+
+    abuse_type_id = abuse_type_name_to_id.get(args.get('abuse_type', ''))
+    abuse_type_other = args.get('abuse_type_other')
+
+    if abuse_type_id is None:
+        raise DemistoException('Bitcoin Abuse: invalid type of abuse, please insert a correct abuse type')
+
+    if abuse_type_id == OTHER_ABUSE_TYPE_ID and abuse_type_other is None:
+        raise DemistoException('Bitcoin Abuse: abuse_type_other is mandatory when abuse type is other')
+
+    return _ReportAddressParams(
+        api_token=api_key,
+        address=args.get('address', ''),
+        abuse_type_id=abuse_type_id,
+        abuse_type_other=abuse_type_other,
+        abuser=args.get('abuser', ''),
+        description=args.get('description', '')
+    )
+
+
+def report_address_command(client: BitcoinAbuseClient, args: Dict) -> str:
+    """
+    Reports a bitcoin abuse to Bitcoin Abuse integration
+
+    Args:
+        client: BitcoinAbuseClient  used to post abuse to the api
+        args: Dict
+
+    Returns:
+        'bitcoin address (address reported) by abuser (abuser reported) was
+        reported to BitcoinAbuse API' if http request was successful'
+    """
+    report_address_params: _ReportAddressParams = _build_report_address_params(args, client.api_key)
+    response = client.report_address(report_address_params)
+    if response.get('success') is True:
+        return f'Bitcoin address {report_address_params.address} by abuse bitcoin user {report_address_params.abuser}' \
+               f' was reported to BitcoinAbuse API'
+    else:
+        return f'bitcoin report address did not succeed: response: {response}'
+
+
+def main() -> None:
+    command = demisto.command()
+    params = {k: v for k, v in demisto.params().items() if v is not None}
+    args = demisto.args()
+
+    verify_certificate = not demisto.params().get('insecure', False)
+    proxy = params.get('proxy', False)
+    api_key = params.get('api_key', '')
+    have_fetched_first_time = demisto.getIntegrationContext().get('have_fetched_first_time', False)
+    first_feed_interval_url_prefix = params.get('initial_fetch_interval', '30d')
+
+    demisto.debug(f'Bitcoin Abuse: Command being called is {demisto.command()}')
+
+    client = BitcoinAbuseClient(
+        verify=verify_certificate,
+        proxy=proxy,
+        api_key=api_key)
+    try:
+
+        if command == 'test-module':
+            download_params = _DownloadParams(client.api_key)
+            client.download_csv(download_params)
+
+            return_results('ok')
+
+        elif command == 'fetch-indicators':
+            feed_url_prefix = build_fetch_indicators_url_prefix(have_fetched_first_time, first_feed_interval_url_prefix)
+            fetch_indicators(params, feed_url_prefix)
+
+        elif command == 'bitcoin-report-address':
```

Comment:
```suggestion
        elif command == 'bitcoinabuse-report-address':
```

Reply:
the command in the design was named bitcoin-report-address
https://github.com/demisto/etc/issues/29462
still want me to change the command name? or talk to Elior about it?

Reply:
@EliorKedar What do you think? I think it is better to go by `bitcoinabuse`